ee/tabby-webserver/src/service/cron/db.rs

```diff
@@ -0,0 +1,58 @@
+use anyhow::Result;
+use tokio_cron_scheduler::Job;
+use tracing::error;
+
+use crate::{schema::JobRunResult, service::db::DbConn};
+
+pub async fn refresh_token_job(db_conn: DbConn) -> Result<Job> {
+    // job is run every 2 hours
+    let job = Job::new_async("0 0 1/2 * * * *", move |_, _| {
+        let db_conn = db_conn.clone();
+        Box::pin(async move {
+            let res = db_conn.delete_expired_token().await;
+            if let Err(e) = res {
+                error!("failed to delete expired token: {}", e);
+            }
+        })
+    })?;
+
+    Ok(job)
+}
+
+pub async fn repository_job(db_conn: DbConn, job_name: String, schedule: &str) -> Result<Job> {
+    let job = Job::new_async(schedule, move |_, _| {
+        let job_name = job_name.clone();
+        let db_conn = db_conn.clone();
+        Box::pin(async move {
+            // run command as a child process
+            let start_time = chrono::Utc::now();
+            let exe = std::env::current_exe().unwrap();
+            let output = tokio::process::Command::new(exe)
+                .arg(&format!("job::{}", &job_name))
+                .output()
+                .await;
+            let Ok(output) = output else {
+                error!("`{}` failed: {:?}", &job_name, output.unwrap_err());
+                return;
+            };
+            let finish_time = chrono::Utc::now();
+
+            // save run result to db
+            let run = JobRunResult {
+                id: 0,
+                job_name,
+                start_time,
+                finish_time,
+                success: output.status.success(),
+                stdout: String::from_utf8_lossy(&output.stdout).to_string(),
+                stderr: String::from_utf8_lossy(&output.stderr).to_string(),
```

Comment:
it'll be really nice to have stdout / stderr result being streammed update to database (e.g every lines trigger an update db operations.)

OK to do this in a following up PR

Reply:
Please help create a ticket for future improvement.